huaweicloud/services/eip/resource_huaweicloud_vpc_eip.go

```diff
@@ -147,395 +215,467 @@ func ResourceVpcEIPV1() *schema.Resource {
-			if err != nil {
-				return fmtp.Errorf("error creating networking v2 client: %s", err)
-			}
-			bandwidth := bandwidthsv2.Bandwidth{
+func updateEipBandwidth(vpcV1Client *golangsdk.ServiceClient, config *config.Config, d *schema.ResourceData) error {
+	old, new := d.GetChange("bandwidth")
+	oldRaw := old.([]interface{})
+	newRaw := new.([]interface{})
+	// Bandwidth blocks are required and must be present.
+	oldMap := oldRaw[0].(map[string]interface{})
+	newMap := newRaw[0].(map[string]interface{})
+
+	bandwidthId := oldMap["id"].(string)
+	if d.Get("charging_mode").(string) == "prePaid" {
+		vpcV2Client, err := config.NetworkingV2Client(config.GetRegion(d))
+		if err != nil {
+			return fmt.Errorf("error creating VPC v2.0 client: %s", err)
+		}
+		updateOpts := bandwidthsv2.UpdateOpts{
+			Bandwidth: bandwidthsv2.Bandwidth{
 				Name: newMap["name"].(string),
 				Size: newMap["size"].(int),
-			}
-			extendParam := &bandwidthsv2.ExtendParam{
+			},
+			ExtendParam: &bandwidthsv2.ExtendParam{
 				IsAutoPay: common.GetAutoPay(d),
-			}
-			updateOpts := bandwidthsv2.UpdateOpts{
-				Bandwidth:   bandwidth,
-				ExtendParam: extendParam,
-			}
-			logp.Printf("[DEBUG] Bandwidth Update Options: %#v", updateOpts)
-
-			order, err := bandwidthsv2.Update(networkingV2Client, eIP.BandwidthID, updateOpts)
-			if err != nil {
-				return fmtp.Errorf("error updating bandwidth: %s", err)
-			}
+			},
+		}
+		log.Printf("[DEBUG] Bandwidth Update Options: %#v", updateOpts)
 
-			orderData, ok := order.(bandwidthsv2.PrePaid)
-			if ok {
-				logp.Printf("[DEBUG] The orderData is: %#v", orderData)
-				// wait for order success
-				bssClient, err := config.BssV2Client(config.GetRegion(d))
-				if err != nil {
-					return fmtp.Errorf("error creating BSS v2 client: %s", err)
-				}
-				if err := orders.WaitForOrderSuccess(bssClient, int(d.Timeout(schema.TimeoutCreate)/time.Second), orderData.OrderID); err != nil {
-					return err
-				}
-			}
-		} else {
-			var updateOpts bandwidths.UpdateOpts
-			updateOpts.Size = newMap["size"].(int)
-			updateOpts.Name = newMap["name"].(string)
-			logp.Printf("[DEBUG] Bandwidth Update Options: %#v", updateOpts)
+		order, err := bandwidthsv2.Update(vpcV2Client, bandwidthId, updateOpts)
+		if err != nil {
+			return fmt.Errorf("error updating bandwidth: %s", err)
+		}
 
-			_, err = bandwidths.Update(networkingClient, eIP.BandwidthID, updateOpts).Extract()
+		if orderData, ok := order.(bandwidthsv2.PrePaid); ok {
+			log.Printf("[DEBUG] The orderData is: %#v", orderData)
+			// Wait for order success.
+			bssClient, err := config.BssV2Client(config.GetRegion(d))
 			if err != nil {
-				return fmtp.Errorf("error updating bandwidth: %s", err)
+				return fmt.Errorf("error creating BSS v2 client: %s", err)
+			}
+			if err := orders.WaitForOrderSuccess(bssClient, int(d.Timeout(schema.TimeoutCreate)/time.Second), orderData.OrderID); err != nil {
```

Comment:
```suggestion
			if err := orders.WaitForOrderSuccess(bssClient, int(d.Timeout(schema.TimeoutCreate)/time.Second), orderData.OrderID); err != nil {
```
			if err := orders.WaitForOrderSuccess(bssClient, int(d.Timeout(schema.TimeoutUpdate)/time.Second), orderData.OrderID); err != nil {

Reply:
done.